sdk/storage/azblob/service/client.go

```diff
@@ -82,6 +83,32 @@ func NewClientFromConnectionString(connectionString string, options *ClientOptio
 	return NewClientWithNoCredential(parsed.ServiceURL, options)
 }
 
+// GetUserDelegationCredential obtains a UserDelegationKey object using the base ServiceURL object.
+// OAuth is required for this call, as well as any role that can delegate access to the storage account.
+func GetUserDelegationCredential(serviceURL string, ctx context.Context, info generated.KeyInfo, timeout *int32, requestID *string) (*UserDelegationCredential, error) {
+	url, err := blob.ParseURL(serviceURL)
+	if err != nil {
+		return nil, err
+	}
+	pl := runtime.NewPipeline(exported.ModuleName, exported.ModuleVersion, runtime.PipelineOptions{}, nil)
+	sc := generated.NewServiceClient(serviceURL, pl)
+	opts := GetUserDelegationKeyOptions{
+		RequestID: requestID,
+		Timeout:   timeout,
+	}
+	udk, err := sc.GetUserDelegationKey(ctx, info, &opts)
+	if err != nil {
+		return nil, err
+	}
+
+	return NewUserDelegationCredential(url.Host, udk.UserDelegationKey), nil
+}
+
+func (s Client) GetUserDelegationKey(ctx context.Context, keyInfo KeyInfo, options *GetUserDelegationKeyOptions) GetUserDelegationKeyResponse {
```

Comment:
Will customers ever directly need to access a `GetUserDelegationKeyResponse`?  What I'm wondering is if this method and `GetUserDelegationCredential` should be combined into a single `GetUserDelegationCredential` API.